File: templates/Uwp/_composition/MVVMLight/Project.SplitView.TabbedNav.Page.Settings/ViewModels/ShellViewModel_postaction.cs
﻿//{**
//This code block adds the logic to handle SettingsItem in NavigationView control from ViewModel.
//**}
//{[{
using Param_RootNamespace.Views;
//}]}
namespace Param_RootNamespace.ViewModels
{
    public class ShellViewModel : ViewModelBase
    {
        private void OnItemInvoked(WinUI.NavigationViewItemInvokedEventArgs args)
        {
            //{[{
            if (args.IsSettingsInvoked)
            {
                NavigationService.Navigate(typeof(wts.ItemNameViewModel).FullName);
                return;
            }

            //}]}
        }

        private void Frame_Navigated(object sender, NavigationEventArgs e)
        {
            IsBackEnabled = NavigationService.CanGoBack;
            //{[{
            if (e.SourcePageType == typeof(wts.ItemNamePage))
            {
                Selected = _navigationView.SettingsItem as WinUI.NavigationViewItem;
                return;
            }

            //}]}
        }
    }
}

File: templates/Uwp/_composition/MVVMLight/Project.SplitView.TabbedNav.Page.Settings._VB/ViewModels/ShellViewModel_postaction.vb
﻿'{**
'This code block adds the logic to handle SettingsItem in NavigationView control from ViewModel.
'**}
'{[{
Imports Param_RootNamespace.Views
'}]}

Namespace ViewModels

    Public Class ShellViewModel
        Inherits ViewModelBase

        Private Sub OnItemInvoked(args As WinUI.NavigationViewItemInvokedEventArgs)
            '{[{
            If args.IsSettingsInvoked Then
                NavigationService.Navigate(GetType(wts.ItemNameViewModel).FullName)
                Return
            End If

            '}]}
        End Sub

        Private Sub Frame_Navigated(sender As Object, e As NavigationEventArgs)
            IsBackEnabled = NavigationService.CanGoBack
            '{[{
            If e.SourcePageType = GetType(wts.ItemNamePage) Then
                Selected = TryCast(_navigationView.SettingsItem, WinUI.NavigationViewItem)
                Return
            End If

            '}]}
        End Sub
    End Class
End Namespace
